httpcore/_backends/asyncio.py

```diff
@@ -120,7 +120,11 @@ async def start_tls(
                 timeout=timeout.get("connect"),
             )
 
-        stream_reader.set_transport(transport)
+        # Initialize the protocol, so the reader is made aware of being tied to
+        # a TLS connection.
```

Comment:
I always find the asyncio protocol/transport stuff _very_ confusing, but this comment seems the wrong way around? This change is to make the _protocol_ aware of the TLS connection.

Reply:
Ah, well yeah you're right. I'm confused too 😆